File: arm.64/langs/go/src/boot.go
package main

import "os"

func boot(targetDir string) []*Cabi {
	var engines []*Cabi
	if envLib := os.Getenv("PAT_LIB"); envLib != "" {
		if c := bind(envLib); c != nil {
			engines = append(engines, c)
			return engines
		}
	}

	files := find(targetDir)
	for _, file := range files {
		if c := bind(file); c != nil {
			engines = append(engines, c)
		}
	}
	return engines
}

File: arm.64/langs/kotlin/src/boot.kt
package pat

import java.lang.foreign.Arena

fun boot(targetDir: String? = null, arena: Arena): List<Cabi> {
    val engines = mutableListOf<Cabi>()
    val envLib = System.getenv("PAT_LIB")
    if (!envLib.isNullOrEmpty()) {
        val c = bind(envLib, arena)
        if (c != null) {
            engines.add(c)
            return engines
        }
    }

    val files = find(targetDir)
    for (file in files) {
        val c = bind(file, arena)
        if (c != null) {
            engines.add(c)
        }
    }
    return engines
}
